stdlib.sh

```diff
@@ -26,6 +25,92 @@ direnv_config_dir=${XDG_CONFIG_DIR:-$HOME/.config}/direnv
 # algorithm and so it won't be re-exported.
 export DIRENV_IN_ENVRC=1
 
+__store_env_strictness() {
+  set +o | grep 'pipefail\|nounset\|errexit' > "/tmp/env_strictness.$$.$1"
+}
+
+__restore_env_strictness() {
+  local file
+  file="/tmp/env_strictness.$$.$1"
+  if [ -f "$file" ]; then
+    # shellcheck disable=SC1090
+    source "$file"
+    rm -f "$file"
+  fi
+}
+
+# Usage: strict_env [COMMAND...]
+#
+# Turns on shell execution strictness. This will force the .envrc
+# evaluation context to exit immediately if:
+#
+# - any command in a pipeline returns a non-zero exit status that is
+#   not otherwise handled as part of `if`, `while`, or `until` tests,
+#   return value negation (`!`), or part of a boolean (`&&` or `||`)
+#   chain.
+# - any variable that has not explicitly been set or declared (with
+#   either `declare` or `local`) is referenced.
+#
+# If followed by a command-line, the strictness applies for the duration
+# of the command.
+#
+# Example:
+#
+#    strict_env
+#    has curl
+#
+#    strict_env has curl
+strict_env() {
```

Comment:
Since there are two uses here, I recommend to early-exit when there are zero arguments.
```suggestion
strict_env() {
  if [[ $# -eq 0 ]]; then
    set -euo pipefail
    return
  fi
```
With the current implementation the __store_env_strictness is called in that first case and the temp file left around on the filesystem.

This would duplicate the `set -o errexit -o nounset -o pipefail` line but I don't think we can do better.